server/routes/api/schema.ts

```diff
@@ -14,7 +15,7 @@ export const ProsemirrorSchema = z.custom<TProsemirrorData>((val) => {
   try {
     const node = Node.fromJSON(schema, val);
     node.check();
-    return true;
+    return !!ProsemirrorHelper.toPlainText(node, schema).trim();
```

Comment:
You can use `isEmptyData` method for this which will avoid having to convert the entire document to text

Reply:
Well, that actually allows whitespaces, so opted for `toPlainText`.

Reply:
I'm not sure if there's a better way than to traverse the entire doc 🤔 

Well, maybe we can halp and break out upon spotting the first non-empty "plain-textified" node?

Reply:
> Well, that actually allows whitespaces

Are you sure? If so it would be fine to update that method behavior.

Reply:
`isEmptyData` is almost there already, if the document isn't a single completely empty paragraph node then it's not empty basically. No need to traverse further

Reply:
>if the document isn't a single completely empty paragraph node then it's not empty basically

```
{
  "data": {
    "type": "doc",
    "content": [
      {
        "type": "paragraph"
      },
      {
        "type": "paragraph"
      },
      {
        "type": "paragraph"
      },
      {
        "type": "paragraph",
        "content": [
          {
            "type": "text",
            "text": "              "
          }
        ]
      }
    ]
  }
}
```

☝️ _Technically_ non-empty, but would be an empty comment _practically_. Therefore, need to traverse till we find a non-empty(trimmed) "plain-textified" node.